File: java/409. Longest Palindrome/Solution.java
class Solution {
    public int longestPalindrome(String s) {

        Map<Character, Integer> freq = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            freq.put(s.charAt(i), freq.getOrDefault(s.charAt(i), 0) + 1);
        }

        int length = 0;
        boolean oddFound = false;

        for (int occurrence : freq.values()) {
            if (!oddFound) oddFound = occurrence % 2 == 1;
            length += (occurrence / 2) * 2;
        }
        if (oddFound) length++;
        return length;
    }
}

File: python/409. Longest Palindrome/solution.py
class Solution:
    def longestPalindrome(self, s: str) -> int:

        freq = {}
        length = 0
        for letter in s:
            freq[letter] = freq.get(letter, 0) + 1

        for value in freq.values():
            length += int(value / 2) * 2
            if (length % 2 == 0 and value % 2 == 1):
                length += 1

        return length